src/Coriolis/non_traditional_beta_plane.jl

```diff
@@ -0,0 +1,80 @@
+"""
+    NonTraditionalBetaPlane{FT} <: AbstractRotation
+
+A Coriolis implementation that accounts for the latitudinal variatino of both
+the locally vertical and the locally horizontal components of the rotation vector.
+Traditionally (see [`BetaPlane`](@ref)) only the locally vertical component is
+accounted for.
+
+This implementation is based off of section 5 of Dellar (2011). It conserve energy,
+angular momentum, and potential vorticity.
+
+References
+==========
+Dellar, P. (2011). Variations on a beta-plane: Derivation of non-traditional
+    beta-plane equations from Hamilton's principle on a sphere. Journal of
+    Fluid Mechanics, 674, 174-195. doi:10.1017/S0022112010006464
+"""
+struct NonTraditionalBetaPlane{FT} <: AbstractRotation
+    fz :: FT
+    fy :: FT
+    β  :: FT
+    γ  :: FT
+end
+
+"""
+    NonTraditionalBetaPlane(FT=Float64;
+        fz=nothing, fy=nothing, β=nothing, γ=nothing,
+        rotation_rate=Ω_Earth, latitude=nothing, radius=R_Earth)
+
+The user may directly specify `fz`, `fy`, `β`, and `γ`, or the three parameters
+`rotation_rate`, `latitude`, and `radius` that specify the rotation rate and
+radius of a planet, and the central latitude (where y = 0) at which the
+non-traditional `β`-plane approximation is to be made.
+
+By default, the `rotation_rate` and planet `radius` is assumed to be Earth's.
+"""
+function NonTraditionalBetaPlane(FT=Float64;
+        fz=nothing, fy=nothing, β=nothing, γ=nothing,
+        rotation_rate=Ω_Earth, latitude=nothing, radius=R_Earth)
+
+    Ω, φ, R =  rotation_rate, latitude, radius
+
+    use_f = !all(isnothing.((fz, fy, β, γ))) && isnothing(latitude)
+    use_planet_parameters = !isnothing(latitude) && all(isnothing((fz, fy, β, γ)))
+
+    if !xor(use_f, use_planet_parameters)
+        throw(ArgumentError("Either the keywords fz, fy, β, and γ must be specified, " *
+                            "*or* all of rotation_rate, latitude, and radius."))
+    end
+
+    if use_planet_parameters
+        fz =  2Ω*sind(φ)
+        fy =  2Ω*cosd(φ)
+        β  =  2Ω*cosd(φ)/R
+        γ  = -4Ω*sind(φ)/R
+    end
+
+    return NonTraditionalBetaPlane{FT}(fz, fy, β, γ)
+end
+
+@inline two_Ωʸ(P, y, z) = P.fy * (1 -  z/P.R) + P.γ * y
+@inline two_Ωᶻ(P, y, z) = P.fz * (1 + 2z/P.R) + P.β * y
+
+@inline two_Ωʸw_minus_two_Ωᶻv(i, j, k, grid, coriolis, U) =
```

Comment:
Looks good. A comment that this function is subsequently interpolated from Cell to Face in `x` might be helpful for future generations.

Reply:
Good point, added.